hazelcast/src/main/java/com/hazelcast/client/impl/connection/tcp/TcpClientConnectionManager.java

```diff
@@ -488,7 +509,12 @@ private boolean doConnectToCandidateCluster(CandidateClusterContext context, boo
                 for (Member member : memberList) {
                     checkClientActive();
                     triedAddressesPerAttempt.add(member.getAddress());
-                    Connection connection = connect(member, o -> getOrConnectToMember((Member) o, switchingToNextCluster));
+                    if (connectionProcessListener != ClientConnectionProcessListener.NOOP) {
+                        connectionProcessListener.attemptingToConnectToAddress(translate(member.getAddress()));
```

Comment:
Two problems there
- We don't always connect to `member.getAddress`, see `DefaultAddressProvider#translate(Member)`
- `translate` could throw

Reply:
1. Use member for translation if possible
2. Added exception handling